runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -234,7 +234,15 @@ private static Object handleReadDataCorruption(CorfuPayloadMsg<ReadResponse> msg
     }
 
 
-
+    /**
+     * Asynchronously write to the logging unit.
+     * @param payload   The log data to write to the logging unit.
+     * @return          A CompletableFuture which will complete with the WriteResult once the
+     *                  write completes.
+     */
+    public CompletableFuture<Boolean> write(ILogData payload) {
+        return router.sendMessageAndGetCompletable(CorfuMsgType.WRITE.payloadMsg(new WriteRequest(payload)));
```

Comment:
I did check the StreamLog code and it does not take into account that if an address already has hole fill data and the  new write request is for a hole fill, returning success would be a good optimization but it throws OverwriteException.
Something to consider as hole fills might have a tendency to come from many clients concurrently.